service/src/main/java/com/ericsson/eiffel/remrem/generate/config/ErLookUpConfig.java

```diff
@@ -0,0 +1,75 @@
+/*
+    Copyright 2019 Ericsson AB.
+    For a full list of individual contributors, please see the commit history.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+package com.ericsson.eiffel.remrem.generate.config;
+
+import javax.annotation.PostConstruct;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+
+import ch.qos.logback.classic.Logger;
+
+@Profile("!integration-test")
+@Configuration
+@Component("erlookup")public class ErLookUpConfig {
+
+    Logger log = (Logger) LoggerFactory.getLogger(ErLookUpConfig.class);
+
+    @Value("${erlookup.url}")
+    private String erURL ;
+    @Value("${erlookup.generate.enabled}")
+    private String erLookUpEnabled;
+
+    private boolean erLookUpCheck;
+
+    public String getErURL() {
+        return erURL;
+    }
+
+    public void setErURL(String erURL) {
+        this.erURL = erURL;
+    }
+
+    public String getErLookUpEnabled() {
+        return erLookUpEnabled;
+    }
+
+    public void setErLookUpEnabled(String erLookUpEnabled) {
+        this.erLookUpEnabled = erLookUpEnabled;
+    }
+    @PostConstruct
+    public void checkAndLoadErLookConfigurationProperties() {
+        erLookUpCheck = Boolean.parseBoolean(erLookUpEnabled);
+        log.info("Checking whether Er configurations enabled or not");
+        if(erLookUpCheck){
+            if(!erURL.isEmpty()){
+            log.info("Er LookUp enabled");
+            setErURL(erURL);
+            log.info("Configured Er lookup URL : " +getErURL());
+            }
+            else{
+                log.info("Enabled ER lookUp configuration but not provided Er URL");
+                log.info("stopping the service.......");
+                System.exit(0);
```

Comment:
In a service do you use System.exit(0) ? I think we need to throw interrupted exception.

It is not a good practice to shutdown the application using System.exit

Reply:
Modified code as suggested